### PR TITLE
No spam sub function

### PR DESCRIPTION
**Pull request by oxarbitrage**
fix object subscription issues.

### CODE REVIEW

**Comment by vikramrajkumar** [comment]
@svk31 Can you review this to make sure everyone in #231 is on the same page?

**Comment by svk31** [comment]
I think there are issues with limit and call orders in this version. Because you never get subscribed to orders, you don't get notified when they changed. `get_full_accounts` should subscribe to all the orders of the account being fetched.

File: libraries/app/database_api.cpp [hunk 1]
```diff
@@ -1772,108 +1794,100 @@ vector<blinded_balance_object> database_api_impl::get_blinded_balances( const fl
...
-         obj = _db.find_object( id );
-         if( obj )
+         const object* obj = nullptr;
+         if( force_notify || is_subscribed_to_item(id) )
          {
-            updates.emplace_back( obj->to_variant() );
-         }
-         else
-         {
-            updates.emplace_back(id); // send just the id to indicate removal
```
Comment (svk31) [inline libraries/app/database_api.cpp]: This is important, removals should only broadcast the id of the object, not the full object.

File: libraries/app/database_api.cpp [hunk 2]
```diff
@@ -1772,108 +1794,100 @@ vector<blinded_balance_object> database_api_impl::get_blinded_balances( const fl
...
 {
-   vector<variant>    updates;
-   map< pair<asset_id_type, asset_id_type>,  vector<variant> > market_broadcast_queue;
+   handle_object_changed(false, ids, 
+      std::bind(&object_database::find_object, &_db, std::placeholders::_1)
+   );
+}
 
-   for(auto id : ids)
+void database_api_impl::handle_object_changed(bool force_notify, const vector<object_id_type>& ids, std::function<const object*(object_id_type id)> find_object)
```
Comment (svk31) [inline libraries/app/database_api.cpp]: Here's a possible fix for removed objects, but you can probably make it more elegant. Notice how you don't even need to perform the object lookup though in the case of removed objects:

```
void database_api_impl::handle_object_changed(bool force_notify, bool is_removal, const vector<object_id_type>& ids, std::function<const object*(object_id_type id)> find_object)
{
   if( _subscribe_callback )
   {
      vector<variant> updates;

      for(auto id : ids)
      {
         const object* obj = nullptr;
         if( force_notify && is_removal)
         {
             updates.emplace_back( id );
         }
         else if( force_notify || is_subscribed_to_item(id) )
         {
            obj = find_object(id);
            if( obj )
            {
                updates.emplace_back( obj->to_variant() );
            }
         }
      }

      broadcast_updates(updates);
   }
```

**Comment by kenCode-de** [comment]
@elmato @svk31 Can we merge this one now?

**Comment by svk31** [comment]
No, we need a commit fixing the above mentioned issue first.

**Comment by oxarbitrage** [comment]
the last changes from @elmato  regarding this issue can be found at:

https://github.com/elmato/bitshares-2/tree/no-spam-sub-function

it is 3 commits ahead of the pull request 234.

https://github.com/elmato/bitshares-2/commits/no-spam-sub-function.

i am testing the branch now, @svk31 if you can test as well from your side will be great.

**Pull request closed without merging**